server/zanata-war/src/main/java/org/zanata/action/ProjectHome.java

```diff
@@ -768,6 +785,11 @@ public String persist() {
             return null;
         }
         getInstance().setSlug(getInputSlugValue());
+        if (!isValidName(getInstance().getName())) {
+            facesMessages.addGlobal(SEVERITY_ERROR,
+                    "This project name is must be alphanumeric");
```

Comment:
"Project name must be alphanumeric" - also, probably should be a Messages object.